dlpy/network.py

```diff
@@ -752,18 +817,72 @@ def load_weights_from_file_with_labels(self, path, format_type='KERAS'):
             contains the weight table.
         format_type : KERAS, CAFFE
             Specifies the source framework for the weights file
+        data_spec: list of :class:`DataSpec`, optional
+            data specification for input and output layer(s)            
+        label_file_name: string, optional
+            Fully qualified path to CSV file containing user-defined 
+            classification labels.  If not specified, ImageNet labels assumed.
 
         '''
         cas_lib_name, file_name = caslibify(self.conn, path, task='load')
 
-        from dlpy.utils import get_imagenet_labels_table
-        label_table = get_imagenet_labels_table(self.conn)
-
-        self._retrieve_('deeplearn.dlimportmodelweights', model=self.model_table,
-                        modelWeights=dict(replace=True, name=self.model_name + '_weights'),
-                        formatType=format_type, weightFilePath=file_name, caslib=cas_lib_name,
-                        labelTable=label_table);
-
+        if (label_file_name):
+            from dlpy.utils import get_user_defined_labels_table
+            label_table = get_user_defined_labels_table(self.conn, label_file_name)
+        else:
+            from dlpy.utils import get_imagenet_labels_table
+            label_table = get_imagenet_labels_table(self.conn)
+            
+        if (data_spec):
+            # see if dataspecs supported by action
+            from io import StringIO
```

Comment:
for StringIO we might need seomthing like this:
# Exception handler for reading StringIO text data into memory
try:
    from StringIO import StringIO
except ImportError:
    from io import StringIO

Reply:
I can add the try/except block as you suggest.